avocado/utils/multipath.py

```diff
@@ -162,6 +162,39 @@ def get_path_status(disk_path):
                     return(paths['dm_st'], paths['dev_st'], paths['chk_st'])
 
 
+def fail_path(path):
+    """
+    failing the individual paths
+    :param disk_path: disk path. Example: sda, sdb.
+    :return: True or False
+    """
+    def is_failed():
+        path_stat = get_path_status(path)
+        if path_stat[0] == 'failed' and path_stat[2] == 'faulty':
+            return True
+        return False
+
+    cmd = 'multipathd -k"fail path %s"' % path
+    if process.system(cmd) == 0:
+        return wait.wait_for(is_failed, timeout=10) or False
+
+
+def reinstate_path(path):
+    """
+    reinstating the individual paths
+    :param disk_path: disk path. Example: sda, sdb.
+    :return: True or False
+    """
+    def is_reinstated():
+        path_stat = get_path_status(path)
+        if path_stat[0] == 'active' and path_stat[2] == 'ready':
+            return True
+        return False
+    cmd = 'multipathd -k"reinstate path %s"' % path
+    if process.system(cmd) == 0:
+        return wait.wait_for(is_reinstated, timeout=10) or False
+
+
```

Comment:
Hello @Naresh-ibm, this module is becoming more and more powerful, which is great. On the other hand I'm not really able to test it. I tried using `scsi_debug` to try it, but failed to set it properly. Would it be in your powers to write a unittest, something like `selftests/functional/test_lv_utils.py`? It'd significantly improve the maintainability of this util.

Reply:
Hi @ldoktor I will certainly work on it. But there are other script which are depending on this function used by us regularly. So it will great help if you can merge it, if you feel everything is fine.